Add Parsing of tensor constants

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #75124
* __->__ #75119
* #75108

Add support for parsing Tensor constants like Double(4, 4) ... by initializing random tensors. This makes saving IR and then parsing it lossy, so I have it toggled as default not on, but is useful in cases like repro-ing Fusions with tensor constants post-freezing.

cc @Krovatkin

Differential Revision: [D35373999](https://our.internmc.facebook.com/intern/diff/D35373999)

Commit:
Add Parsing of tensor constants

[ghstack-poisoned]

ZolotukhinM commented on torch/csrc/jit/ir/irparser.h:
```diff
@@ -13,17 +14,27 @@ struct Graph;
 /** \brief Parse IR from \p STR constructing the corresponding IR in\ GRAPH.
  *
- * \p VMAP is filled with String to Value pairs allowing to index Values in the
+ * O\p VMAP is filled with String to Value pairs allowing to index Values in the
```
Typo.

ZolotukhinM reviewed (approved):
This is supernice, thanks! A couple of remarks below:

eellison commented on torch/csrc/jit/ir/irparser.cpp:
```diff
@@ -117,17 +128,21 @@ VarWithType IRParser::parseVarWithType(bool allow_optional) {
+  bool continue_parsing;
+  do {
+    if (L.cur().kind == TK_IDENT) {
+      name += L.expect(TK_IDENT).text();
+    } else {
+      name += L.expect(TK_NUMBER).text();
     }
-    return name;
-  } else {
-    return L.expect(TK_NUMBER).text();
-  }
+    continue_parsing = false;
+    if (L.nextIf('.')) {
+      continue_parsing = true;
+      name += '.';
+    }
+  } while (continue_parsing);
+  return name;
```
Oop, didnt mean to include that, it's in the next PR https://github.com/pytorch/pytorch/pull/75124

Commit:
Update on "Add Parsing of tensor constants"


Add support for parsing Tensor constants like Double(4, 4) ... by initializing random tensors. This makes saving IR and then parsing it lossy, so I have it toggled as default not on, but is useful in cases like repro-ing Fusions with tensor constants post-freezing.

cc @Krovatkin 

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35373999).

Commit:
Update on "Add Parsing of tensor constants"


Add support for parsing Tensor constants like Double(4, 4) ... by initializing random tensors. This makes saving IR and then parsing it lossy, so I have it toggled as default not on, but is useful in cases like repro-ing Fusions with tensor constants post-freezing.

cc @Krovatkin

Differential Revision: [D35373999](https://our.internmc.facebook.com/intern/diff/D35373999)

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35373999).
